packages/admin/cms-admin/src/pages/pageTree/useExtractPages.tsx

```diff
@@ -0,0 +1,270 @@
+import { gql, useApolloClient } from "@apollo/client";
+import { LocalErrorScopeApolloContext } from "@comet/admin";
+import { readClipboard } from "@comet/blocks-admin";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+import slugify from "slugify";
+
+import { useContentScope } from "../../contentScope/Provider";
+import { GQLPageQuery, GQLPageQueryVariables, GQLUpdatePageMutation, GQLUpdatePageMutationVariables } from "../../documents/types";
+import { GQLPageTreePageFragment, GQLUpdatePageNodeMutation, GQLUpdatePageNodeMutationVariables } from "../../graphql.generated";
+import { useLocale } from "../../locale/useLocale";
+import { usePageTreeContext } from "./usePageTreeContext";
+
+interface ContentsClipboard {
+    textContents: {
+        [key: string]: string;
+    };
+}
+
+interface ParsedContents {
+    contents: {
+        original: string;
+        replaceWith: string;
+    }[];
+}
+/**
+ * Typeguard to check if an object is a ContentsClipboard Type
+ * @param contentsClipboard
+ */
+function isContentsClipboard(contentsClipboard: ContentsClipboard): contentsClipboard is ContentsClipboard {
+    return (contentsClipboard as ContentsClipboard).textContents !== undefined;
+}
+
+/**
+ * Union return type from `getContentsFromClipboard` function.
+ * The union discriminator `canPaste` returns either a ContentsClipboard data if it could be parsed, otherwise an localized error in form of a ReactNode
+ */
+type GetContentsFromClipboardResponse = { canPaste: true; content: ParsedContents } | { canPaste: false; error: React.ReactNode };
+
+const updatePageNodeMutation = gql`
+    mutation ImportContentsUpdatePageNode($nodeId: ID!, $input: PageTreeNodeUpdateInput!) {
+        updatePageTreeNode(id: $nodeId, input: $input) {
+            id
+            name
+            slug
+        }
+    }
+`;
+
+const transformToSlug = (name: string, locale: string) => {
+    let slug = slugify(name, { replacement: "-", lower: true, locale });
+    // Remove everything except unreserved characters and percent encoding (https://tools.ietf.org/html/rfc3986#section-2.1)
+    // This is necessary because slugify does not remove all reserved characters per default (e.g. "@")
+    slug = slug.replace(/[^a-zA-Z0-9-._~]/g, "");
+    return slug;
+};
+
+interface UseExtractPagesApi {
+    /**
+     * Iterates over passed pages synchronous and extracts contents
+     *
+     * Process:
+     *      1. traverses the tree with top-down strategy
+     *          1a. documents and blocks return their translatable text contents
+     *      2. returns stringified JSON including all extracted text contents
+     *
+     * @param flatPagesTree
+     */
+    prepareForExtraction: (pages: GQLPageTreePageFragment[]) => Promise<string>;
```

Comment:
I'd call this `extractContents`. And it would be better to return the array and not a string, because there might be other use cases where we'd like to work with the raw data